<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SFValueInputDialog</class>
 <widget class="QDialog" name="SFValueInputDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>350</width>
    <height>200</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modify Scalar Field</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  
  <layout class="QVBoxLayout" name="mainLayout">
   <property name="spacing">
    <number>12</number>
   </property>
   <property name="margin">
    <number>12</number>
   </property>
   
   <!-- 信息显示区域 -->
   <item>
    <widget class="QLabel" name="infoLabel">
     <property name="text">
      <string>Scalar Field Information</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="styleSheet">
      <string>QLabel { 
        background-color: #f0f0f0; 
        padding: 10px; 
        border-radius: 5px;
        border: 1px solid #d0d0d0;
      }</string>
     </property>
    </widget>
   </item>
   
   <!-- 分隔线 -->
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   
   <!-- 数值输入区域 -->
   <item>
    <layout class="QFormLayout" name="formLayout">
     <property name="horizontalSpacing">
      <number>20</number>
     </property>
     <item row="0" column="0">
      <widget class="QLabel" name="valueLabel">
       <property name="text">
        <string>New Value:</string>
       </property>
       <property name="styleSheet">
        <string>font-weight: bold;</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QDoubleSpinBox" name="valueSpinBox">
       <property name="toolTip">
        <string>Enter the new scalar field value for selected points</string>
       </property>
       <property name="decimals">
        <number>6</number>
       </property>
       <property name="minimum">
        <double>-999999999.000000000000000</double>
       </property>
       <property name="maximum">
        <double>999999999.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="styleSheet">
        <string>QDoubleSpinBox {
          padding: 5px;
          font-size: 12pt;
        }</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   
   <!-- 弹性空间 -->
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   
   <!-- 按钮区域 -->
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   
  </layout>
 </widget>
 
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SFValueInputDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SFValueInputDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>